elodie/filesystem.py

```diff
@@ -98,8 +98,8 @@ def get_all_files(self, path, extensions=None):
             subclasses = get_all_subclasses(Base)
             for cls in subclasses:
                 extensions.update(cls.extensions)
-
         for dirname, dirnames, filenames in os.walk(path):
+            dirnames[:] = [d for d in dirnames if d not in excludeDir]
             for filename in filenames:
                 # If file extension is in `extensions` then append to the list
                 if os.path.splitext(filename)[1][1:].lower() in extensions:
```

Comment:
Something like this is what I was thinking. I did not try to run this so there maybe errors :).

This takes the set of strings passed into `--exclude` and compiles them into a list of regular expression objects. Then for each file path (including directory) it will skip any which match against the expression.

For your original bug request this should work using `--exclude=".*@eaDir.*"`, I think.
```
# Create a list of compiled regular expressions to match against the file path
exclude_regexes = [re.compile(regex) for regex in excludeDir]
filename_path = os.path.join(dirname, filename)
                 if (
                        os.path.splitext(filename)[1][1:].lower() in extensions and
                        not any(regex.match(filename_path) for regex in exclude_regexes)
                    ):
                        yield filename_path
```